src/execution_profile.rs

```diff
@@ -0,0 +1,74 @@
+use pyo3::prelude::*;
+use scylla::client;
+use std::time::Duration;
+
+use crate::enums::{Consistency, SerialConsistency};
+
+#[pyclass(frozen)]
+#[derive(Clone)]
+pub(crate) struct ExecutionProfile {
+    pub(crate) _inner: client::execution_profile::ExecutionProfile,
+}
+
+#[pymethods]
+impl ExecutionProfile {
+    #[new]
+    #[pyo3(signature = (timeout=None, consistency=None, serial_consistency=None))]
+    pub(crate) fn new(
+        timeout: Option<PyObject>,
+        consistency: Option<PyObject>,
+        serial_consistency: Option<PyObject>,
+    ) -> PyResult<Self> {
+        let mut profile_builder = client::execution_profile::ExecutionProfile::builder();
+
+        if let Some(timeout) = timeout {
+            let secs = Python::with_gil(|py| timeout.extract::<f64>(py))?;
+            if !secs.is_finite() || secs <= 0.0 {
+                return Err(pyo3::exceptions::PyValueError::new_err(
+                    "timeout must be a positive, finite number (in seconds)",
+                ));
+            }
+            let millis = (secs * 1000.0) as u64;
+            profile_builder = profile_builder.request_timeout(Some(Duration::from_millis(millis)));
+        }
```

Comment:
nit: There is `Duration::from_secs_f32`. Btw is that the idiomatic way to pass durations in Python? Is ther nothing like a duration type? It's not obvious to me as a user if the float represents seconds, or milliseconds.